### PR TITLE
feat: ADR-54 oracle for MANA prices

### PR DESCRIPTION
**Pull request by fzavalia**
Created ADR for mana prices being calculated according to MANA/USD rate instead of a fixed amount of MANA

### CODE REVIEW

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Use an Oracle for MANA pricing according to USD rate
+
+## Statement of the problem
+
+Some features require the user to pay a certain amount of MANA for execution. These features include, publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection, and claiming a name, were the user is charged a fixed amount of MANA to claim. There is also the purchasing of third party item slots, also priced in MANA.
+
+Given the price fluctuations of the token, more importantly the increase in price over time, these features have become more expensive for the user to consume, thus discouraging content creators from actually adding content to the platform, hurting Decentraland as a whole on the process.
+
+In the casing of collection publishing, prices are being changed manually each weak so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedius and not completely precise as MANA price might still fluctuate between publication price changes.
```
Comment (LautaroPetaccio) [inline docs/ADR-54-oracle-for-mana-pricing.md:9]:
```suggestion
In the case of publishing a collection, prices are being changed manually each week so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedious and not completely precise as MANA price might still fluctuate between publication price changes.
```

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Solution proposed
```
Comment (LautaroPetaccio) [inline docs/ADR-54-oracle-for-mana-pricing.md:11]:
```suggestion
## Proposed solution
```

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Chainlink MANA/USD Data Feed allows our contracts to obtain the rate of MANA in USD which will be used to calculate the amount of MANA required to perform certain transactions given a previously defined amount of USD.
+
+For more information about Chainlink go check their [homepage](https://chain.link/) or their [whitepaper](https://chain.link/whitepaper).
+
+In summary, Chainlink is a solution in which via Oracles, one can obtain off-chain data in a tamper-proof and still decentralized manner to be used on-chain.
+
+In order to implement these price feeds we will create a new contract `ChainlinkOracle` with a `getRate` function that will the be used by other contracts which require obtaining the rate of MANA/USD to reflect a price accordingly. This `ChainlinkOracle` will receive both a Data Feed (Represented as an [AggregatorV3Interface](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol) by the Chainlink [docs](https://docs.chain.link/docs/get-the-latest-price/#solidity)) and the amount of decimals I want the rate to be returned with.
```
Comment (LautaroPetaccio) [inline docs/ADR-54-oracle-for-mana-pricing.md:19]:
```suggestion
In order to implement these price feeds we will create a new contract `ChainlinkOracle` with a `getRate` function that will be used by other contracts which require obtaining the rate of MANA/USD to reflect a price accordingly. This `ChainlinkOracle` will receive both a Data Feed (Represented as an [AggregatorV3Interface](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol) by the Chainlink [docs](https://docs.chain.link/docs/get-the-latest-price/#solidity)) and the amount of decimals the rate will be returned with.
```

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Use an Oracle for MANA pricing according to USD rate
+
+## Statement of the problem
+
+Some features require the user to pay a certain amount of MANA for execution. These features include, publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection, and claiming a name, were the user is charged a fixed amount of MANA to claim. There is also the purchasing of third party item slots, also priced in MANA.
```
Comment (LautaroPetaccio) [inline docs/ADR-54-oracle-for-mana-pricing.md:5]: What do you think about enumerating the cases?
```suggestion
Some features require the user to pay a certain amount of MANA for execution. These features include:
- Publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection
- Claiming a name, were the user is charged a fixed amount of MANA to claim.
- Purchasing Third Party item slots, which is charged according to the amount of slots bought.
```

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Some features require the user to pay a certain amount of MANA for execution. These features include:
+
+- Publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection
+
+- Claiming a name, where the user is charged a fixed amount of MANA to claim.
+
+- Purchasing Third Party item slots, which are charged according to the number of slots bought.
+
+Given the price fluctuations of the token, more importantly, the increase in price over time, these features have become more expensive for the user to consume, thus discouraging content creators from actually adding content to the platform, hurting Decentraland as a whole on the process.
```
Comment (nachomazzara) [inline docs/ADR-54-oracle-for-mana-pricing.md:13]: I will remove this paragraph and add  just something like

```The community has been used the DAO to define the price in MANA but pegged to the USD dollar```

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+In the case of publishing a collection, prices are being changed manually each week so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedious and not completely precise as MANA price might still fluctuate mid-week.
```
Comment (nachomazzara) [inline docs/ADR-54-oracle-for-mana-pricing.md:15]: This is not 100% true. It has been changed manually from 500 MANA to 100 MANA once. Can you rephrase it?

File: docs/ADR-54-oracle-for-mana-pricing.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+## Proposed solution
+
+Chainlink MANA/USD Data Feed allows our contracts to obtain the rate of MANA in USD. This can be used, given an original price in USD, to calculate how much MANA is required by the user to perform a certain transaction.
```
Comment (nachomazzara) [inline docs/ADR-54-oracle-for-mana-pricing.md:19]: I will add something like `in Ethereum and Polygon chains`

**Review by nachomazzara — CHANGES_REQUESTED** [review]
Excellent!!!

I have left minor comments

File: docs/ADR-54-oracle-for-mana-pricing.md (hunk 5 above)
Comment (fzavalia) [reply docs/ADR-54-oracle-for-mana-pricing.md]: Done

File: docs/ADR-54-oracle-for-mana-pricing.md (hunk 6 above)
Comment (fzavalia) [reply docs/ADR-54-oracle-for-mana-pricing.md]: Done

File: docs/ADR-54-oracle-for-mana-pricing.md (hunk 7 above)
Comment (fzavalia) [reply docs/ADR-54-oracle-for-mana-pricing.md]: Dome

**Review by nachomazzara — APPROVED** [review]
